### PR TITLE
Replace ArgParser with Clikt

### PR DESCRIPTION
**Pull request by kkoser**
Clikt has an apache2 license, which matches with our own licensing

Also added some tests around the new annotation matching parameters

### CODE REVIEW

File: parser/build.gradle [hunk 1]
```diff
@@ -27,15 +27,15 @@ repositories {
 
 dependencies {
     api "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
-    api "com.xenomachina:kotlin-argparser:2.0.7"
+    implementation("com.github.ajalt.clikt:clikt:3.1.0")
```
Comment (drewhannay) [inline parser/build.gradle]:
```suggestion
    implementation "com.github.ajalt.clikt:clikt:3.1.0"
```

**Review by drewhannay — APPROVED** [review]

File: parser/build.gradle (hunk 1 above)
Comment (picklebento) [reply parser/build.gradle]: Do we want to standardize double quotes and single quote as well?

File: parser/src/main/kotlin/com/linkedin/dex/parser/DexParser.kt [hunk 2]
```diff
@@ -21,17 +26,23 @@ import java.util.zip.ZipInputStream
...
-        help = "path to output dir where AllTests.txt file will be saved, if not set output will go to stdout"
+    val outputDir by argument().help("path to output dir where AllTests.txt file will be saved, if not set output will go to stdout"
     ).default("")
 
-    val customAnnotations by parser.adding(
-        "-A", help = "add custom annotation used by tests") { toString() }
+    val customAnnotations: List<String> by option("-A", "--annotation").multiple().help("add custom annotation used by tests")
+
+    override fun run() {
+        println("Got file $apkPath output dir $outputDir, annotations: q${customAnnotations}q")
```
Comment (picklebento) [inline parser/src/main/kotlin/com/linkedin/dex/parser/DexParser.kt]: Will the additional console output break any consuming scripts?

Comment (kkoser) [reply parser/src/main/kotlin/com/linkedin/dex/parser/DexParser.kt]: Oh you're right, i meant to remove this woops

**Review by drewhannay — APPROVED** [review]

**Pull request merged**